src/Scientist/FireAndForgetResultPublisher.cs

```diff
@@ -0,0 +1,58 @@
+﻿using GitHub.Internals;
+using System;
+using System.Threading.Tasks;
+
+namespace GitHub
+{
+    public enum ResultPublisherExceptionBehavior
+    {
+        Rethrow,
+        Ignore
+    }
+
+    public class FireAndForgetResultPublisher : IResultPublisher
+    {
+        private readonly IResultPublisher _publisher;
+        private readonly Action<Exception> _onPublisherException;
+        private readonly ConcurrentSet<Task> _publishingTasks = new ConcurrentSet<Task>();
+        private readonly ResultPublisherExceptionBehavior _exceptionBehavior;
+
+        public FireAndForgetResultPublisher(IResultPublisher publisher, ResultPublisherExceptionBehavior exceptionBehavior, Action<Exception> onPublisherException = null)
+        {
+            _publisher = publisher;
+            _onPublisherException = onPublisherException;
+            _exceptionBehavior = exceptionBehavior;
+        }
+
+        public Task Publish<T, TClean>(Result<T, TClean> result)
+        {
+
+            // Disable the warning, because the task is being tracked on the set, 
+            // and immediately removed upon completion.
+#pragma warning disable CS4014
+            var subTask = Task.Run(async () =>
+            {
+                try
+                {
+                    await _publisher.Publish(result);
+                }
+                catch(Exception ex)
+                {
+                    _onPublisherException?.Invoke(ex);
+                    if(_exceptionBehavior == ResultPublisherExceptionBehavior.Rethrow)
+                        throw;
```

Comment:
Right now, our code requires someone pass in an `_onPublisherException` handler. If we allowed that to be null, then maybe I'd understand also having this option. As it stands, if you want this exception logged, do it in the `_onPublisherException` callback.